server/internal/infrastructure/db/postgres/migration/20250526124117_init.up.sql

```diff
@@ -0,0 +1,125 @@
+CREATE TABLE IF NOT EXISTS round (
+    id VARCHAR PRIMARY KEY,
+    starting_timestamp BIGINT NOT NULL,
+    ending_timestamp BIGINT NOT NULL,
+    ended BOOLEAN NOT NULL,
+    failed BOOLEAN NOT NULL,
+    stage_code INTEGER NOT NULL,
+    txid VARCHAR NOT NULL,
+    connector_address VARCHAR NOT NULL,
+    version INTEGER NOT NULL,
+    swept BOOLEAN NOT NULL,
+    vtxo_tree_expiration BIGINT NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS tx_request (
+    id VARCHAR PRIMARY KEY,
+    round_id VARCHAR NOT NULL,
+    FOREIGN KEY (round_id) REFERENCES round(id)
+);
+
+CREATE TABLE IF NOT EXISTS receiver (
+    request_id VARCHAR NOT NULL,
+    pubkey VARCHAR,
+    onchain_address VARCHAR,
+    amount BIGINT NOT NULL,
+    FOREIGN KEY (request_id) REFERENCES tx_request(id),
+    PRIMARY KEY (request_id, pubkey, onchain_address)
+);
+
+CREATE TABLE IF NOT EXISTS tx (
+    txid VARCHAR PRIMARY KEY,
+    tx TEXT NOT NULL,
+    round_id VARCHAR NOT NULL,
+    type VARCHAR NOT NULL,
+    position INTEGER NOT NULL,
+    tree_level INTEGER,
+    parent_txid VARCHAR,
+    is_leaf BOOLEAN,
+    FOREIGN KEY (round_id) REFERENCES round(id)
+);
+
+CREATE TABLE IF NOT EXISTS vtxo (
+    txid VARCHAR NOT NULL,
+    vout INTEGER NOT NULL,
+    pubkey VARCHAR NOT NULL,
+    amount BIGINT NOT NULL,
+    round_tx VARCHAR NOT NULL,
+    spent_by VARCHAR NOT NULL,
+    spent BOOLEAN NOT NULL,
+    redeemed BOOLEAN NOT NULL,
+    swept BOOLEAN NOT NULL,
+    expire_at BIGINT NOT NULL,
+    created_at BIGINT NOT NULL,
+    request_id VARCHAR,
+    redeem_tx TEXT,
+    PRIMARY KEY (txid, vout),
+    FOREIGN KEY (request_id) REFERENCES tx_request(id)
+);
+
+
+CREATE TABLE IF NOT EXISTS market_hour (
+   id SERIAL PRIMARY KEY,
+   start_time BIGINT NOT NULL,
+   end_time BIGINT NOT NULL,
+   period BIGINT NOT NULL,
+   round_interval BIGINT NOT NULL,
+   updated_at BIGINT NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS virtual_tx (
+    txid VARCHAR PRIMARY KEY,
+    tx TEXT NOT NULL,
+    starting_timestamp BIGINT NOT NULL,
+    ending_timestamp BIGINT NOT NULL,
+    expiry_timestamp BIGINT NOT NULL,
+    fail_reason VARCHAR,
+    stage_code INTEGER NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS checkpoint_tx (
+    txid VARCHAR PRIMARY KEY,
+    tx TEXT NOT NULL,
+    commitment_txid TEXT NOT NULL,
+    is_root_commitment_tx BOOLEAN NOT NULL DEFAULT FALSE,
+    virtual_txid VARCHAR NOT NULL,
+    FOREIGN KEY (virtual_txid) REFERENCES virtual_tx(txid)
+);
+
+CREATE UNIQUE INDEX IF NOT EXISTS idx_round_txid ON round(txid);
```

Comment:
is this still needed? in sqlite we needed this because we don't store the commitment tx in the tx table. Here we actually do, so to get a round via txid we could fetch the record from the tx table and get the round via the round_id foreign key. Wdyt?

Reply:
here again txid from round domain is used in many places in codebase

Reply:
Done